presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/DigestAndPercentileArrayStateFactory.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.util.array.ObjectBigArray;
+import io.airlift.stats.QuantileDigest;
+
+import java.util.List;
+
+import static io.airlift.slice.SizeOf.SIZE_OF_DOUBLE;
+import static java.util.Objects.requireNonNull;
+
+public class DigestAndPercentileArrayStateFactory
+        implements AccumulatorStateFactory<DigestAndPercentileArrayState>
+{
+    @Override
+    public DigestAndPercentileArrayState createSingleState()
+    {
+        return new SingleDigestAndPercentileArrayState();
+    }
+
+    @Override
+    public Class<? extends DigestAndPercentileArrayState> getSingleStateClass()
+    {
+        return SingleDigestAndPercentileArrayState.class;
+    }
+
+    @Override
+    public DigestAndPercentileArrayState createGroupedState()
+    {
+        return new GroupedDigestAndPercentileArrayState();
+    }
+
+    @Override
+    public Class<? extends DigestAndPercentileArrayState> getGroupedStateClass()
+    {
+        return GroupedDigestAndPercentileArrayState.class;
+    }
+
+    public static class GroupedDigestAndPercentileArrayState
+            extends AbstractGroupedAccumulatorState
+            implements DigestAndPercentileArrayState
+    {
+        private final ObjectBigArray<QuantileDigest> digests = new ObjectBigArray<>();
+        private final ObjectBigArray<List<Double>> percentilesArray = new ObjectBigArray<>();
+        private long size;
+        private boolean nonNullDataIngested = false;
```

Comment:
don't need to set this value to false. default unspecified value is false already.